### PR TITLE
[FIX] Some fixes and improvements in l10n_cl module

### PR DESCRIPTION
**Pull request by suv-bmya**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_cl/models/account_move.py [hunk 1]
```diff
@@ -27,20 +27,23 @@ def _get_l10n_latam_documents_domain(self):
         domain = super()._get_l10n_latam_documents_domain()
         if (self.journal_id.l10n_latam_use_documents and
                 self.journal_id.company_id.country_id == self.env.ref('base.cl')):
+            domain += [('active', '=', True)]
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py]: domain active = True not needed

File: addons/l10n_cl/models/account_move.py [hunk 2]
```diff
@@ -27,20 +27,23 @@ def _get_l10n_latam_documents_domain(self):
...
             if self.type in ['in_invoice', 'in_refund']:
                 if self.partner_id.l10n_cl_sii_taxpayer_type == '2':
                     domain += [('code', '=', '71')]
                 return domain
             document_type_ids = self.journal_id.l10n_cl_sequence_ids.mapped('l10n_latam_document_type_id').ids
             domain += [('id', 'in', document_type_ids)]
             if self.partner_id.l10n_cl_sii_taxpayer_type == '3':
-                domain += [('code', 'in', ['35', '38', '39', '41'])]
+                domain += [('code', 'in', {'35', '38', '39', '41'})]
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py]: Still in Odoo we normally prefer [], just to be uniform.

File: addons/l10n_cl/models/account_move.py [hunk 3]
```diff
@@ -27,20 +27,23 @@ def _get_l10n_latam_documents_domain(self):
...
         return domain
 
     @api.constrains('type', 'l10n_latam_document_type_id')
     def _check_invoice_type_document_type(self):
         super()._check_invoice_type_document_type()
         for rec in self.filtered('l10n_latam_document_type_id'):
+            if rec.company_id.country_id != self.env.ref('base.cl'):
+                continue
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py]: 2 times filtering, maybe once.

File: addons/l10n_cl/models/res_partner.py [hunk 4]
```diff
@@ -8,17 +8,23 @@ class ResPartner(models.Model):
     _name = 'res.partner'
     _inherit = 'res.partner'
 
+    def _get_default_l10n_cl_sii_taxpayer_type(self):
+        allowed_company = self._context.get('allowed_company_ids')
+        if not allowed_company:
+            return self._context.get('install_module') == 'l10n_cl' and '1'
+        company = self.env['res.company'].browse(allowed_company[0])
+        return company.country_id == self.env.ref('base.cl') and '1'
```
Comment (jco-odoo) [inline addons/l10n_cl/models/res_partner.py]: I think partners are created without company by default.  Even if the partner is not related to Chile, there is no problem in having a '1'.  So, is this change really needed?

File: addons/l10n_cl/models/res_partner.py [hunk 5]
```diff
@@ -8,17 +8,23 @@ class ResPartner(models.Model):
...
-    l10n_cl_sii_taxpayer_type = fields.Selection(_sii_taxpayer_types,
-        'Taxpayer Types', index=True, default='1',
+    l10n_cl_sii_taxpayer_type = fields.Selection(_sii_taxpayer_types, 'Taxpayer Types', index=True,
+        default=_get_default_l10n_cl_sii_taxpayer_type,
         help='1 - VAT Affected (1st Category) (Most of the cases)\n'
-        '2 - Fees Receipt Issuer (Applies to suppliers who issue fees receipt)\n'
-        '3 - End consumer (only receipts)\n'
-        '4 - Foreigner'
+             '2 - Fees Receipt Issuer (Applies to suppliers who issue fees receipt)\n'
+             '3 - End consumer (only receipts)\n'
```
Comment (jco-odoo) [inline addons/l10n_cl/models/res_partner.py]: '4'?

File: addons/l10n_cl/models/res_partner.py (hunk 4 above)
Comment (Danisan) [reply addons/l10n_cl/models/res_partner.py]: This is needed because in tests (Rubi sent us a video), in a multiloc environment you were creating a partner from a different country to CL, the taxpayer type was established to 1 (remember that the field is being shown in the partner form).

File: addons/l10n_cl/models/res_partner.py (hunk 5 above)
Comment (Danisan) [reply addons/l10n_cl/models/res_partner.py]: ok will add!

File: addons/l10n_cl/models/account_move.py (hunk 1 above)
Comment (suv-bmya) [reply addons/l10n_cl/models/account_move.py]: Removed in 712aeab

File: addons/l10n_cl/models/account_move.py (hunk 3 above)
Comment (suv-bmya) [reply addons/l10n_cl/models/account_move.py]: Improved in https://github.com/odoo/odoo/pull/37038/commits/712aeab2bd59a28cfb8914be85fab4ab6ecfdd1e

File: addons/l10n_cl/models/res_partner.py (hunk 5 above)
Comment (suv-bmya) [reply addons/l10n_cl/models/res_partner.py]: Added in https://github.com/odoo/odoo/pull/37038/commits/712aeab2bd59a28cfb8914be85fab4ab6ecfdd1e

File: addons/l10n_cl/models/res_partner.py (hunk 4 above)
Comment (jco-odoo) [reply addons/l10n_cl/models/res_partner.py]: @Danisan We said that it was no problem to have a '1' everywhere.  And in theory it is, but of course it can be weird for the user if he sees the field.  But this code is going too far.  I would rather hide the field if the country is Chile then (but that is a bit against what we do with all the rest, although if it is only needed for Chilean companies it can make sense)  @ren-odoo

File: addons/l10n_cl/models/account_move.py (hunk 2 above)
Comment (jco-odoo) [reply addons/l10n_cl/models/account_move.py]: @Danisan I think this the cause of one of the errors Rubi encountered.  Our views don't accept domains with {}.

File: addons/l10n_cl/models/res_partner.py (hunk 4 above)
Comment (Danisan) [reply addons/l10n_cl/models/res_partner.py]: @jos-odoo if the criteria are to hide fields there are several fields in the same condition. To hide each one individually we have a lot of work to do. 

Nevertheless, there is a proposal from us with a filter by localization. Did you consider this?

Comment (ren-odoo) [reply addons/l10n_cl/models/res_partner.py]: @jco-odoo Thinking in the multi company/mutlilocalization approach my only concern if we hide the field is the next scenario: 
If a contact is created in an Argentinean company and the Chilean taxpayer type is not displayed or filled with a default value then if at some point the same contact needs to be used on and Chilean company we can face blocking errors on the invoice because the document type depends on this field.  This is might be a corner case but let me know if you think it can affect this decision. 
cc. @Danisan

Comment (ren-odoo) [reply addons/l10n_cl/models/res_partner.py]: Hi @Danisan after analyzing it with Jos we decide that the best approach is the next (this is gonna be the same for l10n_ar so we can have the same standard):
- It's ok to display the field regardless of the partner company. 
- The default value should be empty so the users can fill it only when necessary (it shouldn't represent an issue on the invoice  as we have the preventing message in the invoice when this field is empty for a partner)

Comment (Danisan) [reply addons/l10n_cl/models/res_partner.py]: @jco-odoo @ren-odoo and @suv-bmya fixed in bab4b6b

File: addons/l10n_cl/models/account_move.py (hunk 2 above)
Comment (Danisan) [reply addons/l10n_cl/models/account_move.py]: Fixed by @suv-bmya at 712aeab2
